mypy warns about invalid types for headers

<!-- Summary. -->

Using requests with types-requests 2.33 I had no warnings, but now using requests 2.34.1's in-built type hints I have this new warning in multiple places:

```
error: Argument "headers" to "post" has incompatible type "dict[str, str]"; expected "MutableMapping[str, str | bytes] | None"  [arg-type]
```

This did not issue a warning in v2.34.0.

## Expected Result

No warning.

## Actual Result

Warning issued.

## Reproduction Steps

```python
import requests

def x() -> None:
    headers = {"hi": "there"}
    requests.post("https://google.com", headers=headers)
```

## System Information

    $ python -m requests.help

```json
{
  "chardet": {
    "version": null
  },
  "charset_normalizer": {
    "version": "3.4.7"
  },
  "cryptography": {
    "version": ""
  },
  "idna": {
    "version": "3.15"
  },
  "implementation": {
    "name": "CPython",
    "version": "3.13.5"
  },
  "platform": {
    "release": "6.12.74+deb13+1-amd64",
    "system": "Linux"
  },
  "pyOpenSSL": {
    "openssl_version": "",
    "version": null
  },
  "requests": {
    "version": "2.34.1"
  },
  "system_ssl": {
    "version": "30500050"
  },
  "urllib3": {
    "version": "2.7.0"
  },
  "using_charset_normalizer": true,
  "using_pyopenssl": false
}
```

<!-- This command is only available on Requests v2.16.4 and greater. Otherwise,
please provide some basic information about your system (Python version,
operating system, &c). -->

--- a/src/requests/_types.py
+++ b/src/requests/_types.py
@@ -109,7 +109,7 @@ if TYPE_CHECKING:
         bytes | str | Iterable[bytes | str] | SupportsRead[bytes | str] | None
     )
 
-    HeadersType: TypeAlias = MutableMapping[str, str | bytes] | None
+    HeadersType: TypeAlias = Mapping[str, str | bytes] | None
 
     CookiesType: TypeAlias = RequestsCookieJar | Mapping[str, str]
 

--- a/src/requests/models.py
+++ b/src/requests/models.py
@@ -83,7 +83,6 @@ from .utils import (
 )
 
 if TYPE_CHECKING:
-    from collections.abc import MutableMapping
     from http.cookiejar import CookieJar
 
     from typing_extensions import Self
@@ -311,7 +310,7 @@ class Request(RequestHooksMixin):
 
     method: str | None
     url: _t.UriType | None
-    headers: MutableMapping[str, str | bytes]
+    headers: Mapping[str, str | bytes]
     files: _t.FilesType
     data: _t.DataType
     json: _t.JsonType

